tests/devices/qubit/test_simulate.py

```diff
@@ -1178,3 +1179,68 @@ def test_qinfo_tf(self):
 
         grad5 = grad_tape.jacobian(results[5], phi)
         assert qml.math.allclose(grad5, expected_grads[5])
+
+
+ml_frameworks_list = [
+    "numpy",
+    pytest.param("autograd", marks=pytest.mark.autograd),
+    pytest.param("jax", marks=pytest.mark.jax),
+    pytest.param("torch", marks=pytest.mark.torch),
+    pytest.param("tensorflow", marks=pytest.mark.tf),
+]
+
+
+# pylint:disable=too-few-public-methods
+class TestMidCircuitMeasurements:
+    """Unit tests for simulating mid-circuit measurements."""
+
+    @pytest.mark.parametrize("ml_framework", ml_frameworks_list)
+    @pytest.mark.parametrize(
+        "postselect_mode", [None, "hw-like", "pad-invalid-samples", "fill-shots"]
+    )
+    def test_simulate_one_shot_native_mcm(self, ml_framework, postselect_mode):
+        """Unit tests for simulate_one_shot_native_mcm"""
+
+        with qml.queuing.AnnotatedQueue() as q:
+            qml.RX(np.pi / 4, wires=0)
+            m = qml.measure(wires=0, postselect=0)
+            qml.RX(np.pi / 4, wires=0)
+
+        circuit = qml.tape.QuantumScript(q.queue, [qml.expval(qml.Z(0)), qml.sample(m)], shots=[1])
+
+        rng = np.random.default_rng(1)
+        rng_keys = [np.random.default_rng(x) for x in rng.integers(0, 2**32, size=100)]
```

Comment:
Without seeding the test would give a false positive 5% of the time I believe.

Reply:
Fixing RNGs locally like the above may not be the best practice though, and get us stuck in a local min. I wonder if we should consider something like https://www.nengo.ai/pytest-rng/ instead, and use the seed from that. 

Though, if we have a way to avoid setting the seed without FP that may be better overall. Any ideas?

Reply:
If we let this take over seed generation, and we only have control over the `salt`, we might find ourselves testing 20 different salt values until we find one that allows all tests to pass.

Reply:
I share Lee's idea about fixing RNGs, which may not be the best practice. Maybe using a different statistical test you could avoid testing more than 20 different salt values, but I am not sure. :thinking:

Reply:
Regardless of which statistical test we use, there is always going to be a false positive rate, and with a test case like this where we have 20 different parametrization combinations, a false positive is almost inevitable unless we use seeding. The difference here is whether we seed each individual test case or if we use `pytest-rng` to seed the entire test suite. If we seed each individual test case we can adjust the seed for each test case such that it passes, but if we use `pytest-rng`, our only control over the entire repo is the salt, and I imagine that one salt that enables a set of tests to pass will cause another set of tests to fail, and it can be almost impossible to find a salt that allows all stochastic tests in the test suite to pass simultaneously.

Reply:
I don't think using `pytest-rng` would be useful here. Let's keep the scope of the random seeds to individual tests or test classes. Just a few questions before resolving this conversation: what's the strategy to update the initial seeds? We should provide a way to regularly update these random seeds. Can we only rely on testing the API instead of checking stochastic results?

Reply:
I believe `pytest-rng` provides a way to update all random seeds used across the test suite by changing the salt.